### PR TITLE
feat: send hubspot share data LESQ-484

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: {owa_music_year}

-Sends data to HubSpot from share page on submit.

## How to test

1. Go to https://deploy-preview-2107--oak-web-application.netlify.thenational.academy
any share page, share resources, check HubStop received event
download events should still be captured

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts [hunk 1]
```diff
@@ -60,6 +60,13 @@ const useResourceFormSubmit = (props: UseResourceFormProps) => {
       setTermsInLocalStorage(terms);
     }
     if (props.type === "download") {
+      await downloadLessonResources(
+        slug,
+        downloads as DownloadResourceType[],
+        props.isLegacyDownload,
+      );
+    }
+    if (props.type === "share" || props.type === "download") {
```
Comment (njblore) [inline src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts:69]: does this need to be in an if? aren't they the only options?

Comment (JBR90) [reply src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts:69]: Yep doesn't need to be there

File: src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts [hunk 2]
```diff
@@ -60,30 +60,31 @@ const useResourceFormSubmit = (props: UseResourceFormProps) => {
...
+        ...utmParams,
+        oakUserId: posthogDistinctId ? posthogDistinctId : undefined,
+        schoolName:
+          schoolId === "homeschool" || schoolId === "notListed"
+            ? schoolId
+            : schoolName,
+      },
+    });
+    const hubspotFormResponse = await hubspotSubmitForm({
+      hubspotFormId: hubspotDownloadsFormId,
```
Comment (njblore) [inline src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts:83]: does share go to the same form as downloads?

**Review by jack-skerrett-oak — APPROVED** [review]

File: src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts (hunk 2 above)
Comment (JBR90) [reply src/components/DownloadAndShareComponents/hooks/useResourceFormSubmit.ts:83]: Yeah i questioned this too, but yes it uses the same form on HubSpot.

**Review by njblore — APPROVED** [review]

**Pull request merged**